src/backend/marsha/core/models/video.py

```diff
@@ -240,6 +252,57 @@ def get_source_s3_key(self, stamp=None):
             resource=self.resource_id, video=self.id, stamp=stamp
         )
 
+    def duplicate(self, playlist):
+        """Duplicate a video to another playlist.
+
+        Parameters
+        ----------
+        playlist: Type[models.Playlist]
+            The playlist to which the duplicated video should be attached. It should pre-exist.
+
+        Returns
+        -------
+        models.Video
+            The duplicated video with all its related tracks.
+
+        """
+        # Duplicate the video
+        new_video = Video.objects.create(
+            duplicated_from=self,
+            lti_id=self.lti_id,
+            playlist=playlist,
+            resource_id=self.resource_id,
+            state=READY,
+            title=self.title,
+            uploaded_on=self.uploaded_on,
+        )
+
+        def copy_track(obj):
```

Comment:
here is the purpose is to limitate the scope of this function ? Is it always valid to declare a function inside a function ?

Reply:
I allow myself to do it when I need to iterate on something that only makes sense in the scope of the function. Not sure if it's a best practice.